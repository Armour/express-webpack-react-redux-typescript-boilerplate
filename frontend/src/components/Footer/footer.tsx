import React from 'react';
import { Translation } from 'react-i18next';

export class Footer extends React.Component {
  public render() {
    return (
      <Translation ns='common'>
        {(t) => (
          <footer className='page-footer'>
            <div className='container'>
              <div className='row'>
                <div className='col l6 s12'>
                  <h5 className='white-text'>{t('footer.title')}</h5>
                  <p className='grey-text text-lighten-4'>{t('footer.content')}</p>
                </div>
                <div className='col l3 offset-l3 s12'>
                  <h5 className='white-text'>{t('footer.links')}</h5>
                  <ul>
                    <li><a className='grey-text text-lighten-3' href='#'>{t('footer.linkOne')}</a></li>
                    <li><a className='grey-text text-lighten-3' href='#'>{t('footer.linkTwo')}</a></li>
                    <li><a className='grey-text text-lighten-3' href='#'>{t('footer.linkThree')}</a></li>
                    <li><a className='grey-text text-lighten-3' href='#'>{t('footer.linkFour')}</a></li>
                  </ul>
                </div>
              </div>
            </div>
            <div className='footer-copyright'>
              <div className='container'>
                Â© 2017 Copyright <a className='grey-text text-lighten-4' href='https://github.com/Armour'>Armour</a>
                <a className='grey-text text-lighten-4 right' href='#'>{t('footer.moreLinks')}</a>
              </div>
            </div>
          </footer>
        )}
      </Translation>
    );
  }
}

export default Footer;
