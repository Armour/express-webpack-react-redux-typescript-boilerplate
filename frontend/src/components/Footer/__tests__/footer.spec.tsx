import React from 'react';
import TestRenderer from 'react-test-renderer';

import { Footer } from '../footer';

describe('Footer', () => {
  it('should renders correctly', () => {
    const renderer = TestRenderer.create(
      <Footer />,
    );
    expect(renderer).toMatchSnapshot();
    renderer.unmount();
  });
});
