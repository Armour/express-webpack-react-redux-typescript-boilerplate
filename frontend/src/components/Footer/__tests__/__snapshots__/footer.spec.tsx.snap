// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should renders correctly 1`] = `
<footer
  className="page-footer"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col l6 s12"
      >
        <h5
          className="white-text"
        >
          footer.title
        </h5>
        <p
          className="grey-text text-lighten-4"
        >
          footer.content
        </p>
      </div>
      <div
        className="col l3 offset-l3 s12"
      >
        <h5
          className="white-text"
        >
          footer.links
        </h5>
        <ul>
          <li>
            <a
              className="grey-text text-lighten-3"
              href="#"
            >
              footer.linkOne
            </a>
          </li>
          <li>
            <a
              className="grey-text text-lighten-3"
              href="#"
            >
              footer.linkTwo
            </a>
          </li>
          <li>
            <a
              className="grey-text text-lighten-3"
              href="#"
            >
              footer.linkThree
            </a>
          </li>
          <li>
            <a
              className="grey-text text-lighten-3"
              href="#"
            >
              footer.linkFour
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    className="footer-copyright"
  >
    <div
      className="container"
    >
      Â© 2017 Copyright 
      <a
        className="grey-text text-lighten-4"
        href="https://github.com/Armour"
      >
        Armour
      </a>
      <a
        className="grey-text text-lighten-4 right"
        href="#"
      >
        footer.moreLinks
      </a>
    </div>
  </div>
</footer>
`;
